include/boost/geometry/strategies/spherical/closest_points_cross_track_box_box.hpp

```diff
@@ -0,0 +1,345 @@
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_SPHERICAL_CLOSEST_POINTS_CROSS_TRACK_BOX_BOX_HPP
+#define BOOST_GEOMETRY_STRATEGIES_SPHERICAL_CLOSEST_POINTS_CROSS_TRACK_BOX_BOX_HPP
+
+#include <boost/config.hpp>
+#include <boost/concept_check.hpp>
+#include <boost/mpl/if.hpp>
+#include <boost/type_traits/is_void.hpp>
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/assert.hpp>
+#include <boost/geometry/core/point_type.hpp>
+#include <boost/geometry/core/radian_access.hpp>
+#include <boost/geometry/core/tags.hpp>
+
+#include <boost/geometry/algorithms/detail/assign_box_corners.hpp>
+#include <boost/geometry/algorithms/detail/closest_points/result.hpp>
+
+#include <boost/geometry/strategies/distance.hpp>
+#include <boost/geometry/strategies/concepts/distance_concept.hpp>
+#include <boost/geometry/strategies/spherical/distance_cross_track.hpp>
+#include <boost/geometry/strategies/spherical/distance_cross_track_point_box.hpp>
+
+#include <boost/geometry/util/math.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace closest_points
+{
+
+class cross_track_box_box_generic
+{
+public:
+
+    template <typename Point, typename Box, typename Result>
+    static bool covered_by_box_with_info(Point const& box_corner,
+                                         Box const& other_box,
+                                         Result& result)
+    {
+        if (geometry::covered_by(box_corner, other_box))
+        {
+            result.lon1 = get_as_radian<0>(box_corner);
+            result.lat1 = get_as_radian<1>(box_corner);
+            result.lon2 = result.lon1;
+            result.lat2 = result.lat1;
+            result.distance = 0;
+            return true;
+        }
+        return false;
+    }
+
+    template <typename Box1, typename Box2, typename Result>
+    static bool box_box_overlap(Box1 const& box1,
+                                Box2 const& box2,
+                                Result& result)
+    {
+        //check first if boxes overlap
+        typedef boost::array<typename point_type<Box1>::type, 4> array;
+
+        array bp1;
+        geometry::detail::assign_box_corners_oriented<true>(box1, bp1);
+
+        for (typename array::const_iterator ait = bp1.begin();
+             ait != bp1.end(); ait++)
+        {
+            if (covered_by_box_with_info(*ait, box2, result))
+            {
+                return true;
+            }
+        }
+
+        array bp2;
+        geometry::detail::assign_box_corners_oriented<true>(box2, bp2);
+
+        for (typename array::const_iterator ait = bp2.begin();
+             ait != bp2.end(); ait++)
+        {
+            if (covered_by_box_with_info(*ait, box1, result))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+};
+
+template
+<
+    typename CalculationType = void,
+    typename Strategy = distance::haversine<double, CalculationType>
+>
+class cross_track_box_box
```

Comment:
Same here, could be `spherical_box_box`.

Reply:
I named it like this to be consistent with distance. I prefer to changed them both at once in a new PR.